### ISSUE TITLE
WCAG Wiki Scrolling

### ISSUE THREAD

**Issue by MarjaE2** [w3c/wcag#4736]
Visited:

https://www.w3.org/WAI/GL/wiki/Animation_caused_by_user_interaction

Even with prefers-reduced-motion, it loads with non-scrolling sidebars on both sides, and with an animated sticky header.

The sidebars lead to this during scrolling and/or paging down:

> The issue that doesn’t seem to be covered is when you scroll a page (usually vertically, but sometimes horizontally) and things move at different speeds, termed parallax scrolling. E.g. the background moves more slowly than the foreground to give an impression of depth/distance.

The header leads to 2 jumps in quick succession after scrolling and/or paging down.

**Comment by mbgower** [comment]
@MarjaE2 
I was not able to replicate your experience using Chrome, Firefox or Safari.

1. Could you please explain what you mean by an animated sticky header? Are you referring to the fact the page allows the header section to scroll out of view? 
2. The sidebars have a reduced level or scrolling, so that their contents is in review while the user navigates the page's main content.

Neither of these scrolling effects seem to violate the language of [2.3.3 Animation from Interactions](https://www.w3.org/WAI/WCAG21/Understanding/animation-from-interactions.html). Each appears to fit within the parameters of what I understood, at time of writing the Understanding document, to be the "the essential movement associated with scrolling".

If a user wishes to avoid having some sections of text not scroll while other sections scroll, the page content can be zoomed so the page reflows to a single column and the scrolling effects is applied to all content of the page. (This occurs for me at the second zoom level of 120%.) 

It has been a number of years since this guidance was created, so to the degree you are able, please provide more details and which portions of the scrolling are problematic, and whether the method outlined in the prior paragraph is helpful. 

Asking @nattarnoff to review and provide additional input.

**Comment by patrickhlauke** [comment]
i'd agree that the way that wiki page behaves has nothing to do with parallax scrolling, nor does it contravene any SC (animation from interaction, or otherwise). propose closing this

**Comment by MarjaE2** [comment]
1. Animated sticky header. When I page down, the 1st time I page down, the screen changes 2x in quick sucession, instead of 1x. I tried to make a screen recording, but it didn't have the frame rate to show this.

2. Separately-scrolling columns. If one column scrolls and another doesn't, they do not scroll at the same speed. I know that's increasingly-standard web design, but that doesn't mean it doesn't *hurt*. In my opinion, the best option would be to hide the columns by default, at least when users have prefers-reduced-motion. So other options would be to add more space between columns, and clearer visual separations; there's a faint scrollbar between 2 of the columns, but it could be bolder, and there's no separator between the other 2 columns.

**Comment by alastc** [comment]
> the 1st time I page down, the screen changes 2x in quick sucession

That sounds like a bug, possibly browser and/or screensize specific, I can't replicate that. In wide-screen it does animate slightly as the fixed header slides in, so that's something that should be behind the reduce-motion flag. 

I'm not entirely sure who to ask about the wiki, @kfranqueiro do you know who could edit the CSS of the wiki?

>  If one column scrolls and another doesn't, they do not scroll at the same speed. 

I don't think that counts as animation. Animation is defined as:
> "addition of steps between conditions to create the illusion of movement or to give a sense of a smooth transition"

Making something fixed to the viewport does not in itself make it animation. It would need to be something that does not move at the speed of the scrolling.

**Comment by MarjaE2** [comment]
It hurts, and it causes motion sickness and migraines, because the one part is scrolling or paging past the other. If you want to redefine parallax and redefine animation to exclude these, then you should have *some* category which includes these.

In either case, when the web accessibility guidelines use inaccessible web design, it encourages web designers to keep using inaccessible web design.

P.S. And with the replies, this github page has started doing the same standard thing.

**Comment by patrickhlauke** [comment]
having fixed/sticky elements is not what is generally understood as "parallax scrolling". i'd be interested in seeing research on how sticky elements have the same adverse effect to people with motion sensitivity

**Comment by kfranqueiro** [comment]
I believe all of these issues are a result of default MediaWiki styles of the 2022 version of the Vector skin.

If I understand correctly, the sticky header only manifests to logged-in users, which explains why multiple folks on the thread couldn't replicate it. (See "[Why is the sticky header not showing on a certain page?](https://www.mediawiki.org/wiki/Reading/Web/Desktop_Improvements/Features/Sticky_Header#Why_is_the_sticky_header_not_showing_on_a_certain_page?)")

If you are indeed viewing the wiki while logged in, one way to resolve this is to go to the [Appearance tab under Preferences](https://www.w3.org/WAI/GL/wiki/Special:Preferences#mw-prefsection-rendering), select "Vector legacy (2010)" under the Skin heading, and click Save.

If you are not logged in, each of the persistent sidebars should include a hide button, which can reduce the layout to a single column.

**Comment by mbgower** [comment]
@MarjaE2 
> It hurts, and it causes motion sickness and migraines, because the one part is scrolling or paging past the other. If you want to redefine parallax and redefine animation to exclude these, then you should have some category which includes these.

The "screen changes 2x in quick succession" suggests an animation effect that _should_ be reduced/eliminated when the prefers_reduced_motion CSS media query is invoked (as @alastc stated). In github, there is also a method of disabling animations (Profile>Accessibility>Motion:Disabled). Since I was not able to replicate the animation you describe, I cannot confirm whether that addresses this specific issue. Our task force does not actualy control that page; I am flagging @shawna-slh so she can investigate.

The phenomenon of "sticky" headers or persistent sidebars is not intended to fit within the definition of parallax scrolling. As @patrickhlauke notes, it would be good to gather/identify research. If you have anything, please provide. I'm flagging @nattarnoff, who led the work that was done on vestibular disorder reactions, and can hopefully investigate this for WCAG 3.

In regard to addressing variations in scrolling, every page I've tested has reached a break point at some level of zooming and resolved to a single column, eliminating the experience that is affecting you. Therefore, a mechanism already exists that allows the content to be transformed to a view that should not trigger the physical reaction you describe. @kfranqueiro  described a number of other potential ways of addressing.

> when the web accessibility guidelines use inaccessible web design, it encourages web designers to keep using inaccessible web design.

We can only address challenges when we become aware of them. There was insufficient research in place at the time of WCAG 2.2 to allow us to define a A/AA scope for Animation from Interactions. In other words, we could not determine a threshold for motion testing, as was done with Flashing. I'm not sure how much research has expanded in this area. It's a challenging area to study, given the potential effect on some participants.

What you describe sounds like an _additional_ challenge, and it has significant ramifications for existing site design. As such, we need to gather known information on the problem, and determine the scope, triggers, and sufficient techniques. Note too, that the WCAG 2 Task Force is restricted to making corrections to existing requirements. New requirements, such as this, are intended to be addressed in WCAG 3. I have labelled this issue as such, added it to a [parking lot](https://github.com/w3c/wcag/wiki/WCAG-3-parking-lot-considerations-from-2.x) on this topic, and alerted people focused on that standard.

**Comment by shawna-slh** [comment]
It seems there are a few different issues in this GitHub issue.

@MarjaE2:
- That wiki page was an internal draft. The published document is [Understanding... Animation from Interactions](https://www.w3.org/WAI/WCAG21/Understanding/animation-from-interactions.html). It is not a wiki page, and we do control that interface.
- Likely you don't need to use these wiki pages. However, if you do, please see Ken's guidance above https://github.com/w3c/wcag/issues/4736#issuecomment-3524546528

Others:
- If this is a MediaWiki issue, it could be filled with them.
- If this has to do with W3C's implementation, @kfranqueiro can route it to the right place.
- Thanks for flagging @nattarnoff for further research into the issue for **potentially** addressing it in [Framework for Accessible Specification of Technologies (FAST)](https://w3c.github.io/fast/) and other guidance in the future as appropriate.

**Comment by patrickhlauke** [comment]
Proposed response from the WG: closing this, noting that having sticky/fixed elements is not, to our understanding, a case of parallax scrolling and does not present the same vertigo/disorientation issues that parallax scrolling causes. In the specific case of the wikimedia wiki, fixed sidebars can also be hidden, removing the offending behaviour.

**Comment by MarjaE2** [comment]
From personal experience, it *does* cause the same kinds of motion sickness, hence the report.

**Comment by bruce-usab** [comment]
@MarjaE2 can you propose some changes/additions to [Understanding Animations from Interactions](https://www.w3.org/WAI/WCAG22/Understanding/animation-from-interactions.html)?  Also, I am not sure WCAG3 is on track to address your issue better than WCAG 2.x does — so please followup on http://w3.org/wai/wcag3

**Comment by patrickhlauke** [comment]
outright banning fixed elements would not be possible under 2,x i fear. it would also unintentionally affect other cases where one thing scrolls while another one doesn't, such as carousels (the carousel moves, but the rest of the page doesn't) etc.

suggest this is a concern that should be explored as part of WCAG 3 work

**Comment by bruce-usab** [comment]
I would also note that the [definition](https://www.w3.org/TR/WCAG22/#dfn-motion-animation) is very specific (emphasis added):
> **_addition of steps between conditions_** to create the illusion of movement or to give a sense of a smooth transition

I don‘t think the behavior @MarjaE2 describes meets that definition.  (Which again reinforces for me that this may be something for WCAG3.)

**Comment by patrickhlauke** [comment]
Response from the WCAG 2.x task force: closing this, noting that having sticky/fixed elements is not, to our understanding, a case of parallax scrolling. There is further concern that covering these cases as failures under 2.3.3. Animation from Interactions could lead to unintended further consequences, such as making carousels, or any other situation where some content scrolls while other page content doesn't, failures, which would be a significant expansion of what the original SC intended to cover.

In the specific case of the wikimedia wiki and how to fix this specific problem, fixed sidebars in the wiki can be hidden, removing the offending behaviour. See https://github.com/w3c/wcag/issues/4736#issuecomment-3524546528

**Comment by alastc** [comment]
I can add this feedback to the WCAG 3 sub-group working on animation provisions. 

However, having fixed elements on a page is a very common and useful feature. It also seems odd this is a trigger when a browser (at least on desktops) will always have static areas around the scrolling region. Why isn't this a trigger all the time, for any page that scrolls?

**Comment by MarjaE2** [comment]
> It also seems odd this is a trigger when a browser (at least on desktops) will always have static areas around the scrolling region. Why isn't this a trigger all the time, for any page that scrolls?

(1) It's easier for users to expand browser windows (and most but not all other app windows) than to resize columns, etc. on each web page.

(2) If the window is the whole width of the screen, then the edge of the screen is a wonderful visual separator. If it isn't, the distance is also a sort of visual separator. I haven't usually had trouble with docks, scrollbars, and *narrow* columns at the edge, unless the body text is too close to those edges.

Taking the wiki as an example, the side columns are very wide, and there's hardly any space between columns. Taking github, the right column is even wider, and again there's hardly any space between columns.

(3) Many pages have non-scrolling elements *in front* of scrolling elements.
